dspace-api/src/main/java/org/dspace/handle/PIDConfiguration.java

```diff
@@ -0,0 +1,229 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.handle;
+
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+
+import org.apache.logging.log4j.Logger;
+import org.dspace.content.Community;
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.DspaceObjectClarinService;
+import org.dspace.core.Context;
+import org.dspace.services.ConfigurationService;
+import org.dspace.utils.DSpace;
+import org.springframework.stereotype.Component;
+
+/* Created for LINDAT/CLARIAH-CZ (UFAL) */
+/**
+ * Class encapsulating PIDs configuration.
+ *
+ * @author Michaela Paurikova (michaela.paurikova at dataquest.sk)
+ */
+@Component
+public class PIDConfiguration {
+    /**
+     * log4j logger
+     */
+    private static Logger log = org.apache.logging.log4j.LogManager.getLogger(PIDConfiguration.class);
+    private static PIDConfiguration instance;
+
+    private static final String CLARIN_PID_COMMUNITY_CONFIGURATIONS_KEYWORD = "lr.pid.community.configurations";
+
+    private static Map<UUID, PIDCommunityConfiguration> pidCommunityConfigurations;
+
+    private ConfigurationService configurationService = new DSpace().getConfigurationService();
+
+    private DspaceObjectClarinService dspaceObjectClarinService =
+            ContentServiceFactory.getInstance().getDspaceObjectClarinService();
+
+    private PIDConfiguration() {
+        initialize();
+    }
+
+    /**
+     * Initializes the singleton
+     */
+    private void initialize() {
+        //All configurations are loaded into one array.
+        //New configuration starts after loaded part contains "community".
+        String[] pidCommunityConfigurationsArray = configurationService.getArrayProperty
+                (CLARIN_PID_COMMUNITY_CONFIGURATIONS_KEYWORD);
+
+        if (Objects.isNull(pidCommunityConfigurationsArray)) {
+            return;
+        }
+
+        //hashmap for creating PIDCommunityConfiguration
+        Map<String, String> map = new HashMap<String, String>();
+        pidCommunityConfigurations = new HashMap<UUID, PIDCommunityConfiguration>();
+        //exists minimally one configuration, so first community is added to map not in cycle
+        String[] keyValue = pidCommunityConfigurationsArray[0].split("=");
```

Comment:
what if there is no "=" in `pidCommunityConfigurationsArray[0]`?

Reply:
Added:
- if ArrayUtils.isEmpty(pidCommunityConfigurationsArray) return null;
- if cannot load property return RuntimeException.